numpy/core/src/multiarray/einsum.dispatch.c.src

```diff
@@ -255,23 +262,13 @@ static void
     #if @unroll_by@ == 4
         const int vstepx4 = vstep * 4;
         for (; count >= vstepx4; count -= vstepx4, data0 += vstepx4, data1 += vstepx4, data_out += vstepx4) {
-            /**begin repeat3
-             * #i = 0, 1, 2, 3#
-             */
-            npyv_@sfx@ a@i@ = npyv_@ld@_@sfx@(data0 + vstep * @i@);
-            npyv_@sfx@ b@i@ = npyv_@ld@_@sfx@(data1 + vstep * @i@);
-            npyv_@sfx@ c@i@ = npyv_@ld@_@sfx@(data_out + vstep * @i@);
-            /**end repeat3**/
-            /**begin repeat3
-             * #i = 0, 1, 2, 3#
-             */
-            npyv_@sfx@ abc@i@ = npyv_muladd_@sfx@(a@i@, b@i@, c@i@);
-            /**end repeat3**/
-            /**begin repeat3
-             * #i = 0, 1, 2, 3#
-             */
-            npyv_@st@_@sfx@(data_out + vstep * @i@, abc@i@);
-            /**end repeat3**/
+            for (int i = 0; i < 4; i++) {
```

Comment:
You may need `#pragma GCC ivdep` to allow gcc to unroll this. I'd check the assembly to make sure.